swan-lake/learn/mocking.md

```diff
@@ -0,0 +1,338 @@
+---
+layout: ballerina-left-nav-pages-swanlake
+title: Mocking
+description: Learn how to use Ballerina's built-in mocking API provided by the test to test modules
+ independent from other modules and external endpoints.
+keywords: ballerina, programming language, testing, mocking
+permalink: /swan-lake/learn/mocking/
+active: mocking
+redirect_from:
+  - /swan-lake/learn/mocking
+  - /swan-lake/learn/mocking/
+---
+
+# Mocking
+
+Mocking is useful to control the behavior of functions and objects to control the communication with other modules and external endpoints. A mock can be created by defining return values or replacing the entire object or function with a user-defined equivalent. This feature will help you to test the Ballerina code independently from other modules and external endpoints.
+
+
+## Mocking Objects
+
+The test module provides capabilities to mock an object for unit testing. This allows you to control the behavior of object member functions and values of member fields via stubbing or replacing the entire object with a user-defined equivalent. This feature will help you to test the Ballerina code independently from other modules and external endpoints.
+
+Mocking of objects can be done in two ways.
+
+
+
+1. Creating a test double (providing an equivalent object in place of the real)
+2. Stubbing the member function or member variable (specifying the behavior of functions and values of variables)
+
+
+### Creating a test double
+
+You can write a custom mock object and substitute it in place of the real object. The custom object should be made structurally equivalent to the real object via the mocking features in the test module.
+
+***Example:***
+
+Let’s make changes to the example we tried out in the Getting Started section. In order to test the getRandomJoke function without actually calling the endpoint, we can use the test:mock function to mock the `get` remote function of the client object.
```

Comment:
```suggestion
Let’s make changes to the example in the Getting Started section. In order to test the` getRandomJoke` function without actually calling the endpoint, use the `test:mock` function to mock the `get` remote function of the client object.
```
Need to link to the example in the "Getting Started" section.

Reply:
Added the link and suggested changes in 99c63fb